demo-main.cc

```diff
@@ -328,7 +288,7 @@ class ImageScroller : public ThreadedCanvasManipulator {
         for (int y = 0; y < screen_height; ++y) {
           const Pixel &p = current_image_.getPixel(
                      (horizontal_position_ + x) % current_image_.width, y);
-          offscreen_->SetPixel(x, y, p.red, p.green, p.blue);
+          matrix_->transformer()->Transform(offscreen_)->SetPixel(x, y, p.red, p.green, p.blue);
```

Comment:
the screen_height and width you are iterating through here ... they should be queried as well through the transformer, no ?